src/dao/AlarmDAO.ts

```diff
@@ -551,6 +531,22 @@ export class AlarmDAO extends AbstractDAO<number, OnmsAlarm> {
    */
   private getDetailsPage(alarm: number|OnmsAlarm): string {
       const alarmId = (typeof(alarm) === 'number' ? alarm : alarm.id);
-      return this.http.server.resolveURL(`alarm/detail.htm`, {id: alarmId});
+      return this.getServer().resolveURL(`alarm/detail.htm`, {id: alarmId});
+  }
+
+  /**
+   * Handle response errors and automatically log "ticketing not enabled" responses.
+   * @param err the HTTP result error
+   */
+  private handleError(err: OnmsResult<OnmsAlarm>): void {
+    if (err.code === 501) {
+      try {
```

Comment:
Can we try to check if server is set before actually trying to get it, instead of hoping it will succeed?

Reply:
Yeah, we should probably rewrite that handling, but this is just a refactor to deduplicate some code.  I intend to do a future pass updating/cleaning up the "capability" stuff, but it's out of scope for getting this merged, IMHO.